---
title: "Culex_OBP_EDA_03052018"
author: "Megan Fritz"
date: "March 5, 2018"
output: github_document
---
###The goal of this work was to examine the divergence in levels of OBP expression in the heads of above- and below-ground Culex pipiens mosquitoes.  These mosquitoes tend to have differing host preferences (see Fritz et. al 2015), where above-ground mosquitoes are more aviphilic and below-ground mosquitoes tend toward mammalophilicity. My primary question was which OBPs, if any, show differences in expression levels and thereby may be likely to be involved in host preference.  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Loading in data sets and libraries.

```{r Loading data sets and libraries, results = "hide"}

library(sciplot)

#use this to specify the path to your data file
setwd("~/Desktop/CulexRNAseq/OBP_analysis/data")
genes_detected <- read.table("all_genes_nocutoff_norm_counts.txt", header = T)
Leal_OBPs <- read.csv("Leal2011_QuinqOBPs_sd01.csv", header = T)

#just pulling out relevant columns from Leal_OBPs
sub_Leal_OBPs <- Leal_OBPs[,c(1:3)]

merged_Fritz_OBP_dataset <- merge(genes_detected, sub_Leal_OBPs, by.x = "ID", by.y = "VectorBase_ID")
merged_Fritz_OBP_dataset$sumCounts <- rowSums(merged_Fritz_OBP_dataset[,c(2:17)])
head(merged_Fritz_OBP_dataset$sumCounts)#checking that rowSums worked
```

###Because some OBPs are more widely expressed than others, I first wanted to understand how many were detected in my dataset.  I did that but summing the number of reads that aligned to each OBP across all treatment groups. 

```{r Examining subsets of OBP data, results = "show"}
#getting only OBPs with certain number of reads aligned
sub_OBP_1 <- subset(merged_Fritz_OBP_dataset, sumCounts>=1)
NROW(sub_OBP_1)

sub_OBP_10 <- subset(merged_Fritz_OBP_dataset, sumCounts>=10)
NROW(sub_OBP_10)

sub_OBP_100 <- subset(merged_Fritz_OBP_dataset, sumCounts>=100)
NROW(sub_OBP_100)

```

##I then reshaped my dataframe from wide to long format, which allowed me to use sciplot to visualize the relationships between OBP expression and treatment group.

```{r Reshaping full OBP data, results = "show"}
sample_names <- names(sub_OBP_10[2:17])
sub_OBP_10_reshaped <- reshape(sub_OBP_10, 
                         varying = list(2:17),
                         direction="long",
                         idvar= "ID",
                         times=sample_names,
                         v.names="Norm_Read_Count",
                         new.row.names=NULL)

names(sub_OBP_10_reshaped)[names(sub_OBP_10_reshaped) == 'time'] <- 'Strain'
sub_OBP_10_reshaped$Rep <- sub_OBP_10_reshaped$Strain
sub_OBP_10_reshaped$Strain = substr(sub_OBP_10_reshaped$Strain,1,nchar(sub_OBP_10_reshaped$Strain)-1)
sub_OBP_10_reshaped$Rep = substr(sub_OBP_10_reshaped$Rep,nchar(sub_OBP_10_reshaped$Rep),nchar(sub_OBP_10_reshaped$Rep))

#sanity check - reformatting
sub_OBP_10_reshaped_CPIJ001365 <- subset(sub_OBP_10_reshaped, ID == "CPIJ001365")
sum(sub_OBP_10_reshaped_CPIJ001365$Norm_Read_Count)
sub_OBP_10_reshaped_CPIJ001365$sumCounts

sub_OBP_10_reshaped_CPIJ014525 <- subset(sub_OBP_10_reshaped, ID == "CPIJ014525")
sum(sub_OBP_10_reshaped_CPIJ014525$Norm_Read_Count)
sub_OBP_10_reshaped_CPIJ014525$sumCounts #that's good news

```
