---
title: "Culex_GO_Analysis_Level_3"
author: "Megan Fritz"
date: "September 7, 2018"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Plots for Gene Ontology Classification.

#####Loading in data sets and libraries.
```{r Loading data sets and libraries, results = "show"}

setwd("~/Desktop/CulexRNAseq")

#Getting the top 14 MF categories
DGC_MF_grVpa <- read.table("./data/GO_level3Dist_data_analysis_09102018/gravidVparous/gravidVparous_GODist_level3_first3colsMF.txt", sep = "\t", header = T)
DGC_MF_paVma <- read.table("./data/GO_level3Dist_data_analysis_09102018/parousVmales/parousVmales_GODist_level3_first3cols_MF.txt", sep = "\t", header = T)
DGC_MF_paVpip <- read.table("./data/GO_level3Dist_data_analysis_09102018/parousVpipiens/parousVpipiens_GODist_level3_first3cols_MF.txt", sep = "\t", header = T)
DGC_MF_grVpip <- read.table("./data/GO_level3Dist_data_analysis_09102018/gravidVpipiens/gravidVpipiens_GODist_level3_first3cols_MF.txt", sep = "\t", header = T)

top_14_MF <- list(DGC_MF_grVpa[c(1:14),3], DGC_MF_paVma[c(1:14),3], DGC_MF_paVpip[c(1:14),3], DGC_MF_grVpip[c(1:14),3])

tot_data_MF <- list(DGC_MF_grVpa[c(1:nrow(DGC_MF_grVpa)),3], DGC_MF_paVma[c(1:nrow(DGC_MF_paVma)),3], DGC_MF_paVpip[c(1:nrow(DGC_MF_paVpip)),3], DGC_MF_grVpip[c(1:nrow(DGC_MF_grVpip)),3])


#Getting the top 14 BP categories
DGC_BP_grVpa <- read.table("./data/GO_level3Dist_data_analysis_09102018/gravidVparous/gravidVparous_GODist_level3_first3cols_BP.txt", sep = "\t", header = T)
DGC_BP_paVma <- read.table("./data/GO_level3Dist_data_analysis_09102018/parousVmales/parousVmales_GODist_level3_first3cols_BP.txt", sep = "\t", header = T)
DGC_BP_paVpip <- read.table("./data/GO_level3Dist_data_analysis_09102018/parousVpipiens/parousVpipiens_GODist_level3_first3cols_BP.txt", sep = "\t", header = T)
DGC_BP_grVpip <- read.table("./data/GO_level3Dist_data_analysis_09102018/gravidVpipiens/gravidVpipiens_GODist_level3_first3cols_BP.txt", sep = "\t", header = T)

top_14_BP <- list(DGC_BP_grVpa[c(1:14),3], DGC_BP_paVma[c(1:14),3], DGC_BP_paVpip[c(1:14),3], DGC_BP_grVpip[c(1:14),3])

tot_data_BP <- list(DGC_BP_grVpa[c(1:nrow(DGC_BP_grVpa)),3], DGC_BP_paVma[c(1:nrow(DGC_BP_paVma)),3], DGC_BP_paVpip[c(1:nrow(DGC_BP_paVpip)),3], DGC_BP_grVpip[c(1:nrow(DGC_BP_grVpip)),3])


#Getting the top 14 CC categories
DGC_CC_grVpa <- read.table("./data/GO_level3Dist_data_analysis_09102018/gravidVparous/gravidVparous_GODist_level3_first3cols_CC.txt", sep = "\t", header = T)
DGC_CC_paVma <- read.table("./data/GO_level3Dist_data_analysis_09102018/parousVmales/parousVmales_GODist_level3_first3cols_CC.txt", sep = "\t", header = T)
DGC_CC_paVpip <- read.table("./data/GO_level3Dist_data_analysis_09102018/parousVpipiens/parousVpipiens_GODist_level3_first3cols_CC.txt", sep = "\t", header = T)
DGC_CC_grVpip <- read.table("./data/GO_level3Dist_data_analysis_09102018/gravidVpipiens/gravidVpipiens_GODist_level3_first3cols_CC.txt", sep = "\t", header = T)

top_14_CC <- list(DGC_CC_grVpa[c(1:14),3], DGC_CC_paVma[c(1:14),3], DGC_CC_paVpip[c(1:14),3], DGC_CC_grVpip[c(1:14),3])

tot_data_CC <- list(DGC_CC_grVpa[c(1:nrow(DGC_CC_grVpa)),3], DGC_CC_paVma[c(1:nrow(DGC_CC_paVma)),3], DGC_CC_paVpip[c(1:nrow(DGC_CC_paVpip)),3], DGC_CC_grVpip[c(1:nrow(DGC_CC_grVpip)),3])



```

#####Getting MF GO percentages by comparison
```{r getting MF GO percentages, results = "hide"}

tot_MF <- lapply(tot_data_MF, sum)
inv_misc_cat_MF <- lapply(top_14_MF, sum)
misc_cat_MF <- (unlist(tot_MF)) - (unlist(inv_misc_cat_MF))

data2_MF <- as.data.frame(top_14_MF, col.names = c("DGC_MF_grVpa", "DGC_MF_paVma", "DGC_MF_paVpip", "DGC_MF_grVpip"))

Updata_MF <- rbind(data2_MF, misc_cat_MF)

Percentages_MF <- data.frame((t(apply(Updata_MF, 1, "/", unlist(tot_MF)))*100))
Rounded_Percentages_MF <- round(Percentages_MF, 1)

```

#####Getting BP GO percentages by comparison
```{r getting BP GO percentages, results = "hide"}

tot_BP <- lapply(tot_data_BP, sum)
inv_misc_cat_BP <- lapply(top_14_BP, sum)
misc_cat_BP <- (unlist(tot_BP)) - (unlist(inv_misc_cat_BP))

data2_BP <- as.data.frame(top_14_BP, col.names = c("DGC_BP_grVpa", "DGC_BP_paVma", "DGC_BP_paVpip", "DGC_BP_grVpip"))

Updata_BP <- rbind(data2_BP, misc_cat_BP)

Percentages_BP <- data.frame((t(apply(Updata_BP, 1, "/", unlist(tot_BP)))*100))
Rounded_Percentages_BP <- round(Percentages_BP, 1)

```

#####Getting CC GO percentages by comparison
```{r getting CC GO percentages, results = "hide"}

tot_CC <- lapply(tot_data_CC, sum)
inv_misc_cat_CC <- lapply(top_14_CC, sum)
misc_cat_CC <- (unlist(tot_CC)) - (unlist(inv_misc_cat_CC))

data2_CC <- as.data.frame(top_14_CC, col.names = c("DGC_CC_grVpa", "DGC_CC_paVma", "DGC_CC_paVpip", "DGC_CC_grVpip"))

Updata_CC <- rbind(data2_CC, misc_cat_CC)

Percentages_CC <- data.frame((t(apply(Updata_CC, 1, "/", unlist(tot_CC)))*100))
Rounded_Percentages_CC <- round(Percentages_CC, 1)

```

#####First Making Pie Charts for Molecular Function

######Here, I used the following abbreviations for my pie chart labels:
######"structural constituent of ribosome" = "struc const ribosome"
######"serine-type endopeptidase activity", "ser-type endopep act"
######"guanyl-nucleotide exchange factor" = "guan-nuc ex factor"
######"substrate-specific transporter activity" = "sub-spec transport act"

```{r MF GO Figs, results = "show"}

lbls <- c(as.character(DGC_MF_grVpa[c(1:14),2]), "Miscellaneous")
lbls <- gsub("structural constituent of ribosome", "struc const ribosome", lbls)
lbls <- gsub("guanyl-nucleotide exchange factor", "guan-nuc ex factor", lbls)
lbls <- gsub("substrate-specific transporter activity", "sub-spec transport act", lbls)
lbls <- paste(lbls, Rounded_Percentages_MF$DGC_MF_grVpa) # add percents to labels
lbls <- paste(lbls,"%",sep="") # ad % to labels 
grVpa_pie_MF <- pie(Rounded_Percentages_MF$DGC_MF_grVpa, labels =lbls, main = "Top Molecular Function GO Categories for Gravid CAL1 versus Parous CAL1 Comparison", radius = 0.9, col = rainbow(length(lbls)), cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 30)

lbls <- c(as.character(DGC_MF_paVma[c(1:14),2]), "Miscellaneous")
lbls <- gsub("structural constituent of ribosome", "struc const ribosome", lbls)
lbls <- gsub("guanyl-nucleotide exchange factor", "guan-nuc ex factor", lbls)
lbls <- gsub("substrate-specific transporter activity", "sub-spec transport act", lbls)
lbls <- paste(lbls, Rounded_Percentages_MF$DGC_MF_paVma) 
lbls <- paste(lbls,"%",sep="") 
paVma_pie_MF <- pie(Rounded_Percentages_MF$DGC_MF_paVma, labels =lbls, main = "Top Molecular Function GO Categories for Parous CAL1 versus Male CAL1 Comparison", radius = 0.9, col = rainbow(length(lbls)), cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 30)

lbls <- c(as.character(DGC_MF_paVpip[c(1:14),2]), "Miscellaneous")
lbls <- gsub("structural constituent of ribosome", "struc const ribosome", lbls)
lbls <- gsub("serine-type endopeptidase activity", "ser-type endopep act", lbls)
lbls <- gsub("guanyl-nucleotide exchange factor", "guan-nuc ex factor", lbls)
lbls <- gsub("substrate-specific transporter activity", "sub-spec transport act", lbls)
lbls <- paste(lbls, Rounded_Percentages_MF$DGC_MF_paVpip) 
lbls <- paste(lbls,"%",sep="") 
paVpip_pie_MF <- pie(Rounded_Percentages_MF$DGC_MF_paVpip, labels =lbls, main = "Top Molecular Function GO Categories for Parous CAL1 versus IL2 Comparison", radius = 0.9, col = rainbow(length(lbls)), cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 30)

lbls <- c(as.character(DGC_MF_grVpip[c(1:14),2]), "Miscellaneous")
lbls <- gsub("structural constituent of ribosome", "struc const ribosome", lbls)
lbls <- gsub("serine-type endopeptidase activity", "ser-type endopep act", lbls)
lbls <- gsub("guanyl-nucleotide exchange factor", "guan-nuc ex factor", lbls)
lbls <- gsub("substrate-specific transporter activity", "sub-spec transport act", lbls)
lbls <- paste(lbls, Rounded_Percentages_MF$DGC_MF_grVpip) 
lbls <- paste(lbls,"%",sep="") 
grVpip_pie_MF <- pie(Rounded_Percentages_MF$DGC_MF_grVpip, labels =lbls, main = "Top Molecular Function GO Categories for Gravid CAL1 versus IL2 Comparison", radius = 0.9, col = rainbow(length(lbls)), cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 30)

```

#####Biological Process GO Comparison Charts
######Here, I used the following abbreviations for my pie chart labels:
######"regulation of transcription, DNA-templated" = "reg transcript, DNA-temp"
######"carbohydrate metabolic process" = "carb metabol proc"
######"establishment of localization" = "establish localization"
######"single-organism cellular process" = "single-org cell process"

```{r BP GO Figs, results = "show"}

lbls <- c(as.character(DGC_BP_grVpa[c(1:14),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("establishment of localization", "establish localization", lbls)
lbls <- gsub("single-organism cellular process", "single-org cell process", lbls)
lbls <- paste(lbls, Rounded_Percentages_BP$DGC_BP_grVpa) # add percents to labels
lbls <- paste(lbls,"%",sep="") # ad % to labels 
grVpa_pie_BP <- pie(Rounded_Percentages_BP$DGC_BP_grVpa, labels =lbls, main = "Top Biological Process GO Categories for Gravid CAL1 versus Parous CAL1 Comparison", radius = 0.9, cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 120)

lbls <- c(as.character(DGC_BP_paVma[c(1:14),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("carbohydrate metabolic process", "carb metabol proc", lbls)
lbls <- gsub("establishment of localization", "establish localization", lbls)
lbls <- gsub("single-organism cellular process", "single-org cell process", lbls)
lbls <- paste(lbls, Rounded_Percentages_BP$DGC_BP_paVma) 
lbls <- paste(lbls,"%",sep="") 
paVma_pie_BP <- pie(Rounded_Percentages_BP$DGC_BP_paVma, labels =lbls, main = "Top Biological Process GO Categories for Parous CAL1 versus Male CAL1 Comparison", radius = 0.9, cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 120)

lbls <- c(as.character(DGC_BP_paVpip[c(1:14),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("carbohydrate metabolic process", "carb metabol proc", lbls)
lbls <- gsub("establishment of localization", "establish localization", lbls)
lbls <- gsub("single-organism cellular process", "single-org cell process", lbls)
lbls <- paste(lbls, Rounded_Percentages_BP$DGC_BP_paVpip) 
lbls <- paste(lbls,"%",sep="") 
paVpip_pie_BP <- pie(Rounded_Percentages_BP$DGC_BP_paVpip, labels =lbls, main = "Top Biological Process GO Categories for Parous CAL1 versus IL2 Comparison", radius = 0.9,  cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 120)

lbls <- c(as.character(DGC_BP_grVpip[c(1:14),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("carbohydrate metabolic process", "carb metabol proc", lbls)
lbls <- gsub("establishment of localization", "establish localization", lbls)
lbls <- gsub("single-organism cellular process", "single-org cell process", lbls)
lbls <- paste(lbls, Rounded_Percentages_BP$DGC_BP_grVpip) 
lbls <- paste(lbls,"%",sep="") 
grVpip_pie_BP <- pie(Rounded_Percentages_BP$DGC_BP_grVpip, labels =lbls, main = "Top Biological Process GO Categories for Gravid CAL1 versus IL2 Comparison", radius = 0.9, cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 120)

```


#####Cellular Component GO Comparison Charts
######Here, I used the following abbreviations for my pie chart labels:
######"regulation of transcription, DNA-templated" = "reg transcript, DNA-temp"
######"carbohydrate metabolic process" = "carb metabol proc"
######"membrane-bounded organelle" = "membrane-bounded org"
######"non-membrane-bounded organelle" = "non-membrane-bounded org"
######"intrinsic component of membrane" = "intrins comp membr"


```{r CC GO Figs, results = "show"}

col_schema <- c("lavender", "purple", "blue", "deepskyblue", "lightblue", "orange", "brown", "yellow3", "yellow", "lightyellow")

lbls <- c(as.character(DGC_CC_grVpa[c(1:9),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("membrane-bounded organelle", "membrane-bounded org", lbls)
lbls <- gsub("non-membrane-bounded organelle", "non-membrane-bounded org", lbls)
lbls <- gsub("intrinsic component of membrane", "intrins comp membr", lbls)
lbls <- paste(lbls, Rounded_Percentages_CC$DGC_CC_grVpa) # add percents to labels
lbls <- paste(lbls,"%",sep="") # ad % to labels 
grVpa_pie_CC <- pie(Rounded_Percentages_CC$DGC_CC_grVpa, labels =lbls, main = "Top Cellular Component GO Categories for Gravid CAL1 versus Parous CAL1 Comparison", radius = 0.9, cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 35, col= col_schema)

lbls <- c(as.character(DGC_CC_paVma[c(1:9),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("membrane-bounded organelle", "membrane-bounded org", lbls)
lbls <- gsub("non-membrane-bounded organelle", "non-membrane-bounded org", lbls)
lbls <- gsub("carbohydrate metabolic process", "carb metabol proc", lbls)
lbls <- gsub("intrinsic component of membrane", "intrins comp membr", lbls)
lbls <- paste(lbls, Rounded_Percentages_CC$DGC_CC_paVma) 
lbls <- paste(lbls,"%",sep="") 
paVma_pie_CC <- pie(Rounded_Percentages_CC$DGC_CC_paVma, labels =lbls, main = "Top Cellular Component GO Categories for Parous CAL1 versus Male CAL1 Comparison", radius = 0.9, cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 60, col= col_schema)

lbls <- c(as.character(DGC_CC_paVpip[c(1:9),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("membrane-bounded organelle", "membrane-bounded org", lbls)
lbls <- gsub("non-membrane-bounded organelle", "non-membrane-bounded org", lbls)
lbls <- gsub("carbohydrate metabolic process", "carb metabol proc", lbls)
lbls <- gsub("intrinsic component of membrane", "intrins comp membr", lbls)
lbls <- paste(lbls, Rounded_Percentages_CC$DGC_CC_paVpip) 
lbls <- paste(lbls,"%",sep="") 
paVpip_pie_CC <- pie(Rounded_Percentages_CC$DGC_CC_paVpip, labels =lbls, main = "Top Cellular Component GO Categories for Parous CAL1 versus IL2 Comparison", radius = 0.9,  cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 50, col= col_schema)

lbls <- c(as.character(DGC_CC_grVpip[c(1:9),2]), "Miscellaneous")
lbls <- gsub("regulation of transcription, DNA-templated", "reg transcript, DNA-temp", lbls)
lbls <- gsub("membrane-bounded organelle", "membrane-bounded org", lbls)
lbls <- gsub("non-membrane-bounded organelle", "non-membrane-bounded org", lbls)
lbls <- gsub("carbohydrate metabolic process", "carb metabol proc", lbls)
lbls <- gsub("intrinsic component of membrane", "intrins comp membr", lbls)
lbls <- paste(lbls, Rounded_Percentages_CC$DGC_CC_grVpip) 
lbls <- paste(lbls,"%",sep="") 
grVpip_pie_CC <- pie(Rounded_Percentages_CC$DGC_CC_grVpip, labels =lbls, main = "Top Cellular Component GO Categories for Gravid CAL1 versus IL2 Comparison", radius = 0.9, cex = 0.75, cex.main = 0.75, tck=.2, init.angle = 50, col= col_schema)

```
